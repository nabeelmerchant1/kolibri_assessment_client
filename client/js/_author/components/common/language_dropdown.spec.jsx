import React                from 'react';
import { shallow }          from 'enzyme';
import LanguageDropdown     from './language_dropdown';

describe('language_dropdown component', () => {
  let props;
  let result;
  let calledFunc;

  beforeEach(() => {
    calledFunc = false;
    props = {
      updateItem: () => { calledFunc = true; },
    };
    result = shallow(<LanguageDropdown {...props} />);
  });

  it('renders a snapshot of the component', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls the updateItem function onChange', () => {
    expect(calledFunc).toBeFalsy();
    result.find('select').simulate('change', { target: { value: 'Spanglish' } });
    expect(calledFunc).toBeTruthy();
  });
});
