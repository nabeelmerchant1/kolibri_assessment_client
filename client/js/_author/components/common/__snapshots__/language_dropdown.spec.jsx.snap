// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language_dropdown component renders a snapshot of the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-c-dropdown au-c-dropdown--small au-u-ml-md"
>
    <select
        id=""
        name=""
        onChange={[Function]}
        tabIndex="0"
        value={undefined}
    >
        <option
            value="639-2%3AENG%40ISO"
        >
            English
        </option>
        <option
            value="639-2%3AHIN%40ISO"
        >
            Hindi
        </option>
        <option
            value="639-2%3ATEL%40ISO"
        >
            Telugu
        </option>
    </select>
</div>,
  "nodes": Array [
    <div
      className="au-c-dropdown au-c-dropdown--small au-u-ml-md"
>
      <select
            id=""
            name=""
            onChange={[Function]}
            tabIndex="0"
            value={undefined}
      >
            <option
                  value="639-2%3AENG%40ISO"
            >
                  English
            </option>
            <option
                  value="639-2%3AHIN%40ISO"
            >
                  Hindi
            </option>
            <option
                  value="639-2%3ATEL%40ISO"
            >
                  Telugu
            </option>
      </select>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <languageDropdown
        updateItem={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "updateItem": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-c-dropdown au-c-dropdown--small au-u-ml-md"
>
          <select
                    id=""
                    name=""
                    onChange={[Function]}
                    tabIndex="0"
                    value={undefined}
          >
                    <option
                              value="639-2%3AENG%40ISO"
                    >
                              English
                    </option>
                    <option
                              value="639-2%3AHIN%40ISO"
                    >
                              Hindi
                    </option>
                    <option
                              value="639-2%3ATEL%40ISO"
                    >
                              Telugu
                    </option>
          </select>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-c-dropdown au-c-dropdown--small au-u-ml-md"
>
          <select
                    id=""
                    name=""
                    onChange={[Function]}
                    tabIndex="0"
                    value={undefined}
          >
                    <option
                              value="639-2%3AENG%40ISO"
                    >
                              English
                    </option>
                    <option
                              value="639-2%3AHIN%40ISO"
                    >
                              Hindi
                    </option>
                    <option
                              value="639-2%3ATEL%40ISO"
                    >
                              Telugu
                    </option>
          </select>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <languageDropdown
    updateItem={[Function]}
/>,
}
`;
